src/pgrn-normalize.c

```diff
@@ -0,0 +1,59 @@
+#include "pgroonga.h"
+#include "pgrn-compatible.h"
+#include "pgrn-global.h"
+#include "pgrn-groonga.h"
+
+#include <utils/builtins.h>
+
+static grn_ctx *ctx = &PGrnContext;
+
+PGRN_FUNCTION_INFO_V1(pgroonga_normalize);
+
+/**
+ * pgroonga_normalize(target text, normalizerName text) : text
+ */
+Datum
+pgroonga_normalize(PG_FUNCTION_ARGS)
+{
+	text *target;
+	text *normalizerName;
+	grn_obj *normalizer;
+	grn_obj *string;
+	unsigned int lengthInBytes;
+	const char *normalized;
+	text *normalizedText;
+
+	target = PG_GETARG_TEXT_PP(0);
+
+	if (PG_NARGS() == 2)
+	{
+		normalizerName = PG_GETARG_TEXT_PP(1);
+		normalizer = PGrnLookupWithSize(VARDATA_ANY(normalizerName),
+										VARSIZE_ANY_EXHDR(normalizerName),
+										ERROR);
+	}
+	else
+	{
+		normalizer = PGrnLookup(PGRN_DEFAULT_NORMALIZER, ERROR);
+	}
+
+	string = grn_string_open(ctx,
+							 VARDATA_ANY(target),
+							 VARSIZE_ANY_EXHDR(target),
+							 normalizer,
+							 0);
+
+	PGrnCheck("pgroonga: failed to open string");
```

Comment:
`"pgroonga: "` prefix is redundant. Because `PGrnCheck()` adds it.

Reply:
`"failed to open string"` is too general. How about adding more tag such as `"normalize: failed to open normalized string"`?

Reply:
Fixed by f0c4383